### ISSUE TITLE
[BFW-6519] [BUG] mmu3 mk4 crash/restart on toolchange with certain gcodes

### ISSUE THREAD

**Issue by flyingkillerspacepixies** [prusa3d/Prusa-Firmware-Buddy#4362]
### Printer model

MK4, MMU3

### Firmware version

6.1.3

### Upgrades and modifications

MMU3, v6 nozzle adapter, G10 bed

### Printing from...

USB Drive

### Describe the bug

When printing some specific multi-material gcodes, the printer will reliably do the following when reaching a toolchange code

1. attempt a filament change while over the purge tower (shows the message on screen indicating it's swapping filament)
2. suddenly the printer will restart as if the reset button was pressed
3. when it turns back on it will be in an error state with the red screen of death and claim there is a nozzle preheat error
  - given this always happens in the same place on subsequent prints. this error message seems to be a red herring

I've attached an example gcode where this happens relatively quickly (layer 3, after ~10min of printing). I stripped it down while testing to eliminate variables but I can upload the original gcode from prusaslicer if wanted. The second T4 toolchange command is where the crash occurs and I removed everything between that and the post-print gcodes.

I've had this happen at other points in a print (like 3/4's through a multi-hour print multiple times for instance ;P) so it doesn't seem to have anything to do with layer height.

### How to reproduce

Just try to print the given gcode on an mk4 + mmu3.

### Expected behavior

The printer doesn't crash, and instead continues on to finish the print.

### Files

[crashy.gcode.txt](https://github.com/user-attachments/files/18240132/crashy.gcode.txt)

**Comment by CZDanol** [comment]
Okay, thank you. I have created an internal ticket BFW-6519 for this and we will investigate the issue.

**Comment by DRracer** [comment]
@flyingkillerspacepixies 
1. The gcode snippet you provided has a completely incorrect ramming and purging parameters for MK4, please reslice it with the latest PrusaSlicer profiles. It looks like a setting taken from MK3S. That will **NOT** work reliably.
2. I've been unable to reproduce any FW crashes on our MK4+MMU3. That means you are probably experiencing some kind of electrical issue, please get in touch with our Tech Support team - they will help you diagnose your machine.

I'm closing this ticket - it's not a FW problem.

**Comment by DRracer** [comment]
@flyingkillerspacepixies ramming are definitely not the same, I've watched the print. On MK4 the nozzle moves as fast as it can. That was not the case in your original G-code. Also, the purged volume was incorrect and huge overextrusion happened on the purge tower.

I'll try your lizPrez.gcode...

**Comment by DRracer** [comment]
Nope, your slicer profile is incorrect, I can see that from the very first toolchange.

**Comment by DRracer** [comment]
@flyingkillerspacepixies if you still experience electrical problems with your setup, please get in touch with our tech support team. The error screen was IMHO not caused by an incorrect print profile.
